Code/C++/searching_in_sorted_rotated_array.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+If a sorted array is rotated around a random pivot and you don't know the pivot.
+      index - [0,1,2,3,4,5]
+i.e - array - [1,3,5,7,8,9] is a sorted array let pivot = 3(this is the index and here zero-based indexing is used)
+                                                                index - [0,1,2,3,4,5]
+around which the array is rotated then resultant array would be array - [8,9,1,3,5,7]
+
+Now this algorithm is used to find the index of key element(which we need to find) in rotated array in O(logn).
+
+key Observation of Algorithm =>
+ -> if we choose any index and split the array into two then one of the two sub array is sorted.
+
+Test case 1 :
```

Comment:
take these test cases to the bottom of the code above the time and space complexity